src/components/Login/index.tsx

```diff
@@ -0,0 +1,78 @@
+import { jwtDecode } from 'jwt-decode';
+import React, { FC, useEffect, useRef } from 'react';
+import styled from '@emotion/styled';
+import { UserInfo } from '../../App';
+import { Button } from '@mui/material';
+import { useOutsideClick } from './useHandleClickOutside';
+import { useOpenGoogleSignIn } from './useOpenGoogleSignIn';
+
+interface PropTypes {
+  email: string;
+  setUser: React.Dispatch<React.SetStateAction<UserInfo>>;
+  showLogin: boolean;
+  setShowLogin: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const UserContainerStyle = styled.div`
+  position: absolute;
+  right: 0;
+  display: flex;
+  align-items: self-start;
+  justify-content: flex-end;
+`;
+
+const SignInContainer = styled.div`
+  background: white;
+  padding: 32px;
+  border-radius: 4px;
+  box-shadow: 0 3px 10px rgb(0 0 0 / 0.2);
+`;
+
+const Login: FC<PropTypes> = ({ email, setUser, showLogin, setShowLogin }) => {
+  const hideSigninDiv = useOpenGoogleSignIn()
```

Comment:
This variable name is confusing, since you pass both `true` and `false` to method as param. Maybe changing it to `updateSignInVisibility` or sth like that would be better?